packages/core-components/src/lib/unsorted/ContentsList.svelte

```diff
@@ -1,12 +1,34 @@
 <script>
-	import { onMount } from 'svelte';
-	let headers;
-	onMount(() => {
-		headers = document.querySelectorAll('h1.markdown, h2.markdown');
-		// Add ID tags to all the headers in the article
+	import { onMount, onDestroy } from 'svelte';
+
+	let headers = [];
+	let observer;
+
+	function updateLinks() {
+		headers = Array.from(document.querySelectorAll('h1.markdown, h2.markdown'));
 		headers.forEach((header, i) => {
-			header.id = encodeURIComponent(header.innerText + i);
+			// Headers may contain values that change in response to user input, so we create our anchors as just the position on the page.
+			header.id = encodeURIComponent(i + 1);
 		});
+	}
+
+	function observeDocumentChanges() {
```

Comment:
Is this firing any time the DOM changes anywhere? Can we scope this somehow to avoid running a querySelectorAll each time?

Probably isn't a big deal in the grand scheme of things, but could (stress on could, this is probably a nitpick) cause perf issues because it's blocking on the main thread